bot/bot.py

```diff
@@ -44,9 +46,27 @@ def __init__(self, *args, redis_session: RedisSession, **kwargs):
             # Since statsd is UDP, there are no errors for sending to a down port.
             # For this reason, setting the statsd host to 127.0.0.1 for development
             # will effectively disable stats.
-            statsd_url = "127.0.0.1"
+            statsd_url = LOCALHOST
 
-        self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        self.stats = AsyncStatsClient(self.loop, LOCALHOST)
+        self._connect_statsd(statsd_url)
+
+    def _connect_statsd(self, statsd_url: str, retry_after: int = 2, attempt: int = 1) -> None:
+        """Callback used to retry a connection to statsd if it should fail."""
+        if self._statsd_timerhandle and not self._statsd_timerhandle.cancelled:
+            self._statsd_timerhandle.cancel()
```

Comment:
Why is it being cancelled in here too? I don't see how the function can be called again before a previous one is executed.

Reply:
Mostly to ensure it wasn't running before i reassigned the attribute